client-sdk/ts-web/rt/src/mrae.ts

```diff
@@ -1,26 +1,26 @@
 import * as oasis from '@oasisprotocol/client';
-import * as ed from '@noble/ed25519';
+import * as nacl from 'tweetnacl';
 import {sha512_256} from 'js-sha512';
 
 var deoxysii = require('deoxysii');
 
-const BoxKDFTweak = 'MRAE_Box_Deoxys-II-256-128';
+const BOX_KDF_TWEAK = 'MRAE_Box_Deoxys-II-256-128';
 
 /**
  * deriveSymmetricKey derives a MRAE AEAD symmetric key suitable for use with the asymmetric
  * box primitives from the provided X25519 public and private keys.
  */
 
 export function deriveSymmetricKey(publicKey: Uint8Array, privateKey: Uint8Array): Uint8Array {
-    const pmk = ed.curve25519.scalarMult(oasis.misc.toHex(privateKey), oasis.misc.toHex(publicKey));
-    var kdf = sha512_256.hmac.create(BoxKDFTweak);
+    const pmk = nacl.scalarMult(privateKey, publicKey);
+    var kdf = sha512_256.hmac.create(BOX_KDF_TWEAK);
     kdf.update(pmk);
     return oasis.misc.fromHex(kdf.hex());
```

Comment:
why not save a parsing step and use [`arrayBuffer`](https://github.com/emn178/js-sha512/blob/b6c443143f4a32cbd62b18bfc8f984c4b54c97f5/index.d.ts#L24)

```js
return new Uint8Array(kdf.arrayBuffer());
```